osu.Game/Users/User.cs

```diff
@@ -228,13 +239,35 @@ public class UserAchievement
         [JsonProperty("replays_watched_counts")]
         public UserHistoryCount[] ReplaysWatchedCounts;
 
-        public class UserHistoryCount
+        [UsedImplicitly]
+        [JsonExtensionData]
+        private readonly IDictionary<string, JToken> otherProperties = new Dictionary<string, JToken>();
+
+        private readonly Dictionary<RulesetInfo, UserStatistics> statisticsCache = new Dictionary<RulesetInfo, UserStatistics>();
+
+        /// <summary>
+        /// Retrieves the user statistics for a certain ruleset.
+        /// If user is fetched from a <see cref="GetUserRequest"/>,
+        /// this will always return empty instance, use <see cref="Statistics"/> instead.
+        /// </summary>
+        /// <param name="ruleset">The ruleset to retrieve statistics for.</param>
+        // todo: this should likely be moved to a separate UserCompact class at some point.
+        public UserStatistics GetStatisticsFor(RulesetInfo ruleset)
         {
-            [JsonProperty("start_date")]
-            public DateTime Date;
+            if (statisticsCache.TryGetValue(ruleset, out var existing))
+                return existing;
 
-            [JsonProperty("count")]
-            public long Count;
+            return statisticsCache[ruleset] = parseStatisticsFor(ruleset);
+        }
+
+        private UserStatistics parseStatisticsFor(RulesetInfo ruleset)
+        {
+            if (!(otherProperties.TryGetValue($"statistics_{ruleset.ShortName}", out var token)))
+                return new UserStatistics();
+
+            var settings = JsonSerializableExtensions.CreateGlobalSettings();
+            settings.DefaultValueHandling = DefaultValueHandling.Include;
+            return token.ToObject<UserStatistics>(JsonSerializer.Create(settings));
```

Comment:
This isn't how we should be doing this. Figure a way to read the json into a `Dictionary<string, Userstatistics>` instead (where the `string` is the `ShortName`). And remove the cache and everything else.

Reply:
Yeah I was struggling on google looking for a way to do this, so I fell back into using a `JsonExtensionData`, will see if a JSON converter would do.

Reply:
Are you sure a dictionary doesn't just work? I know we've dealt with a similar structure somewhere else before, so it may pay to start by looking at other responses in this project to find a similar case.

Reply:
Well, the problem is that `statistics_{osu/taiko/...}` are direct children of the main `User` json object:
```json
{
    "users": [
        {
            "avatar_url": "https://a.ppy.sh/14210502?1597999788.jpeg",
            "country_code": "SA",
            "default_group": "default",
            "id": 14210502,
            "statistics_fruits": {
                 ...
            },
            "statistics_mania": {
                 ...
            },
            "statistics_osu": {
                 ...
            },
            "statistics_taiko": {
                 ...
            },
        ...
        }
    ]
}
```

So I presume a proper way would be making sometype of custom transformer that deserializes the usual properties to a `User` instance, and then deserialize the statistics and pass them to that instance, which seems to be too much. 

But I've simplified this into parsing the statistics on deserialization (via `OnDeserialized` attribute), which feels pretty good to me than this unecessary caching logic.

---

I would say prehaps osu-web should've included these statistics inside one JSON dictionary there, maybe worth reporting issue there if feasible?

Reply:
yep

Reply:
Done https://github.com/ppy/osu-web/issues/7197, will hold off from pushing until that's addressed.